catboost/libs/model/model_export/cpp_exporter.cpp

```diff
@@ -238,99 +142,131 @@ namespace NCB {
         out << --indent << "}" << '\n';
         out << --indent << "}" << '\n';
         out << --indent << "};" << '\n';
-    };
-
-    void TCatboostModelToCppConverter::WriteModelCatFeatures(const TFullModel& model, const THashMap<ui32, TString>* catFeaturesHashToString) {
-        CB_ENSURE(model.ModelTrees->GetDimensionsCount() == 1, "Export of MultiClassification model to cpp is not supported.");
+    }
 
-        WriteCTRStructs();
-        Out << '\n';
+    void TCatboostModelToCppConverter::WriteApplicator(bool forCatFeatures) {
+        if (forCatFeatures) {
+            Out << NResource::Find("catboost_model_export_cpp_ctr_calcer");
+            Out << '\n';
+            Out << NResource::Find("catboost_model_export_cpp_model_applicator");
+        } else {
+            Out << NResource::Find("catboost_model_export_cpp_model_applicator_without_cat");
+        }
+    }
 
+    void TCatboostModelToCppConverter::WriteModel(bool forCatFeatures, const TFullModel& model, const THashMap<ui32, TString>* catFeaturesHashToString) {
         TIndent indent(0);
         TSequenceCommaSeparator comma;
+
         Out << "/* Model data */" << '\n';
 
-        int binaryFeatureCount = model.ModelTrees->GetEffectiveBinaryFeaturesBucketsCount();
+        int binaryFeatureCount = forCatFeatures
+                ? model.ModelTrees->GetEffectiveBinaryFeaturesBucketsCount()
+                : GetBinaryFeatureCount(model);
 
         Out << indent++ << "static const struct CatboostModel {" << '\n';
-        Out << indent << "CatboostModel() {};" << '\n';
+        Out << indent << "CatboostModel() = default;" << '\n';
         Out << indent << "unsigned int FloatFeatureCount = " << model.GetNumFloatFeatures() << ";" << '\n';
-        Out << indent << "unsigned int CatFeatureCount = " << model.GetNumCatFeatures() << ";" << '\n';
+        if (forCatFeatures) {
+            Out << indent << "unsigned int CatFeatureCount = " << model.GetNumCatFeatures() << ";" << '\n';
+        }
```

Comment:
I think it is better to always have this member so it would be more explicit that the model has 0 categorical features.